### PR TITLE
Fix landscape image issue

### PR DESCRIPTION
**Pull request by luka-nextcloud**
Signed-off-by: Luka Trovic <luka@nextcloud.com>

* Resolves: #1075
`Possibly remove max-height for landscape pictures (ref Jos’ screenshot issue when zooming)`

* Target version: master 

### Summary
Fix landscape pictures position issue.
image
image

### CODE REVIEW

**Comment by jancborchardt** [comment]
The image still has a rather small max-height though, as seen from the padding on the left and right?

The images could get a `max-height: 100vh;` that way they would always fit the screen/viewport height.

**Comment by luka-nextcloud** [comment]
@jancborchardt 
I've checked again, the light green color is the author's background that we need to leave as it is.

File: src/nodes/ImageView.vue [hunk 1]
```diff
@@ -260,12 +260,15 @@ export default {
 
 	.image__view {
 		text-align: center;
-
-		.image__main {
-			max-height: 40vh;
+		img {
+			width: 100%;
```
Comment (juliushaertl) [inline src/nodes/ImageView.vue:264]:
```suggestion
			max-width: 100%;
```

Should still be max-width, otherwise portrait images would be stretched.

**Comment by juliushaertl** [comment]
@jancborchardt The max-height was picked deliberately iirc also in discussion with you at some point to always make the image fit into the visible area. I'd be fine to drop, just want to double check that this is fine with you then. 100vh would not take into account that there is also a menubar visible, so ideally it should be `calc(100vh - 100px)` to take the size without header + menubar

**Comment by jancborchardt** [comment]
> @jancborchardt The max-height was picked deliberately iirc also in discussion with you at some point to always make the image fit into the visible area. I'd be fine to drop, just want to double check that this is fine with you then. 100vh would not take into account that there is also a menubar visible, so ideally it should be `calc(100vh - 100px)` to take the size without header + menubar

No you’re right, the max-height should be kept, ideally with the value you propose. :)

File: src/nodes/ImageView.vue [hunk 2]
```diff
@@ -260,12 +260,15 @@ export default {
...
 		}
 	}
 
+	.image__main {
+		max-height: 80vh;
```
Comment (juliushaertl) [inline src/nodes/ImageView.vue:269]:
```suggestion
		max-height: calc(100vh - 50px - 50px);
```

**Comment by juliushaertl** [comment]
@luka-nextcloud THen with my two code suggestions this would be good :+1:

**Comment by luka-nextcloud** [comment]
@juliushaertl Fixed :)

**Review by juliushaertl — APPROVED** [review]

**Pull request merged**